src/generic/Residue.jl

```diff
@@ -126,6 +126,10 @@ end
 deepcopy_internal(a::AbstractAlgebra.ResElem, dict::IdDict) =
    parent(a)(deepcopy(data(a)))
 
+function characteristic(a::ResRing{T}) where T <: Integer
```

Comment:
Should'nt it be
```suggestion
function characteristic(a::ResRing{T}) where T <: RingElement
```
?

Reply:
No. What I have is correct. You cannot determine the characteristic in general, only when you have an integer.

Reply:
Oh sorry, I must have forgotten to think.